app/src/main/java/com/example/a310_rondayview/FragmentHome.java

```diff
@@ -5,16 +5,113 @@
+import android.widget.TextView;
+import com.bumptech.glide.Glide;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
 
 public class FragmentHome extends Fragment {
 
-    public FragmentHome() {
-        // Required empty public constructor
-    }
+    private List<Event> events = new ArrayList<>();
+    private int currentEventIndex = 0;
+
+    // Views
+    private TextView clubNameTextView;
+    private TextView eventTitleTextView;
+    private TextView locationTextView;
+    private TextView timeTextView;
+    private TextView eventDescriptionTextView;
+    private ImageView eventImageView;
+
+    private ImageView eventClubPFPImageView;
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_home, container, false);
+        View rootView = inflater.inflate(R.layout.fragment_home, container, false);
+
+        // Find the CardView that includes the event container layout
+        View eventCardView = rootView.findViewById(R.id.eventCardView);
+
+        // Find the included layout within the CardView
+        View eventContainer = eventCardView.findViewById(R.id.event_container);
+
+        // Find views within the included layout
+        clubNameTextView = eventContainer.findViewById(R.id.clubNameTextView);
+        eventTitleTextView = eventContainer.findViewById(R.id.eventTitleTextView);
+        locationTextView = eventContainer.findViewById(R.id.locationTextView);
+        timeTextView = eventContainer.findViewById(R.id.dateTimeTextView);
+        eventDescriptionTextView = eventContainer.findViewById(R.id.eventDescriptionTextView);
+        eventImageView = eventContainer.findViewById(R.id.eventImageView);
+        eventClubPFPImageView = eventContainer.findViewById(R.id.profileImageView);
+
+        // Fetch data from Firestore
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        db.collection("events").get().addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                for (QueryDocumentSnapshot document : task.getResult()) {
+                    Event event = new Event(
+                            document.getId(),
+                            document.getString("clubName"),
+                            document.getString("title"),
+                            document.getString("description"),
+                            document.getString("location"),
+                            document.getTimestamp("dateTime"),
+                            document.getString("imageURL"),
+                            document.getString("eventClubProfilePicture")
+                    );
+                    events.add(event);
+                }
```

Comment:
I think it's better if we use the Firestore ```toObject``` method to grab the Event, in case we need to change the fields. Have you tried using that? Otherwise, we can keep this for now and change it once we get the data manager working.

Reply:
Yeah you're so right, I'll implement this change now aswell